core/api/test/integration/app/accounts/update-account-status.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { randomUUID } from "crypto"
+
+import { Accounts } from "@/app"
+
+import { AccountsRepository } from "@/services/mongoose"
+import { createRandomUserAndBtcWallet } from "test/helpers"
+
+describe("updateAccountStatus", () => {
+  it("sets account status (with history) for given user id", async () => {
+    const newWalletDescriptor = await createRandomUserAndBtcWallet()
+    const newAccount = await AccountsRepository().findById(newWalletDescriptor.accountId)
+    if (newAccount instanceof Error) throw newAccount
+
+    const updatedByPrivilegedClientId = randomUUID() as PrivilegedClientId
+
+    let account = await Accounts.updateAccountStatus({
+      accountId: newAccount.id,
+      status: "pending",
```

Comment:
this should throw an error... we should not be able to set again pending status :thinking:

Reply:
well I thought we had that validation.. it is a domain validation error... but probably out of scope